interactive-clap-derive/src/derives/interactive_clap/mod.rs

```diff
@@ -373,36 +192,65 @@ pub fn impl_interactive_clap(ast: &syn::DeriveInput) -> TokenStream {
     }
 }
 
-fn context_scope_for_struct(
-    name: &syn::Ident,
-    context_scope_fields: Vec<proc_macro2::TokenStream>,
-) -> proc_macro2::TokenStream {
-    let interactive_clap_context_scope_for_struct = syn::Ident::new(
-        &format!("InteractiveClapContextScopeFor{}", &name),
-        Span::call_site(),
-    );
-    quote! {
-        pub struct #interactive_clap_context_scope_for_struct {
-            #(#context_scope_fields,)*
-        }
-        impl interactive_clap::ToInteractiveClapContextScope for #name {
-            type InteractiveClapContextScope = #interactive_clap_context_scope_for_struct;
-        }
-    }
-}
+/// these are common methods, reused for both the [structs] and `enums` derives
+pub(super) mod common_methods;
 
-fn context_scope_for_struct_field(field: &syn::Field) -> proc_macro2::TokenStream {
-    let ident_field = &field.ident.clone().expect("this field does not exist");
-    let ty = &field.ty;
-    if !self::methods::fields_with_subcommand::is_field_with_subcommand(field)
-        && !self::methods::fields_with_subargs::is_field_with_subargs(field)
-    {
-        quote! {
-            pub #ident_field: #ty
+/** This module describes [`crate::InteractiveClap`] derive logic in case when [`syn::DeriveInput`]
+is a struct
+
+The structure of produced derive output is as follows, where code blocks are generated by
+submodules with corresponding names:
+
+```rust,ignore
+quote::quote! {
+    #to_cli_trait_block
+    #input_args_impl_block
+    #to_interactive_clap_context_scope_trait_block
+    #from_cli_trait_block
+    #clap_for_named_arg_enum_block
+}
+```
+*/
+pub(crate) mod structs {
+    /// returns the whole result `TokenStream` of derive logic of containing module
+    pub fn token_stream(
+        name: &syn::Ident,
+        cli_name: &syn::Ident,
+        ast: &syn::DeriveInput,
+        fields: &syn::Fields,
+    ) -> proc_macro2::TokenStream {
+        let b1 = to_cli_trait::token_stream(name, cli_name, fields);
+        let b2 = input_args_impl::token_stream(ast, fields);
+        let b3 = to_interactive_clap_context_scope_trait::token_stream(ast, fields);
+        let b4 = from_cli_trait::token_stream(ast, fields);
+        let b5 = clap_for_named_arg_enum::token_stream(ast, fields);
+
+        quote::quote! {
+            #b1
+            #b2
+            #b3
+            #b4
+            #b5
         }
-    } else {
-        quote!()
     }
+
+    #[doc = include_str!("../../../docs/structs_to_cli_trait_docstring.md")]
+    pub(crate) mod to_cli_trait;
```

Comment:
Same here, move them to the top.

Reply:
Can you move the docs back to this file (or to the file itself)? I think it's crucial to keep it here, as it's mostly for internal devs, and it's not pleasant to search for a separate file. 

As a developer, I never build docs for my crate just to understand the logic.

This is pretty important point for me

Reply:
yeap, ok
the thing is really convoluted to understand, that's why the doc. 
as a reference point to return later, when context will be forgotten. 
if it helps me navigate, maybe it'll help someone else, though the latter is unlikely